### PR TITLE
chore(extension): unify saturation related utils

### PR DESCRIPTION
**Pull request by vetalcore**
# Checklist

- [x] JIRA -[LW-9552](https://input-output.atlassian.net/browse/LW-9552)
- [ ] Proper tests implemented
- [ ] Screenshots added.

---

## Proposed solution

Explain how does this PR solves the problem stated in JIRA ticket.
You can also enumerate different alternatives considered while approaching this task.

## Testing

Describe here, how the new implementation can be tested.
Provide link or briefly describe User Acceptance Criteria/Tests that need to be met

## Screenshots

Attach screenshots here if implementation involves some UI changes

[LW-9552]: https://input-output.atlassian.net/browse/LW-9552?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: packages/cardano/src/ui/util/pool-metrics.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const oversaturatedLowerBound = 100;
+export const isOversaturated = (saturation: number): boolean =>
+  inRange(saturation, oversaturatedLowerBound, Number.MAX_SAFE_INTEGER);
+
+type Entries<T> = {
+  [K in keyof T]: [K, T[K]];
+}[keyof T][];
+
+export const getSaturationLevel = (saturation: number): SaturationLevels => {
+  let result = SaturationLevels.Medium;
```
Comment (tommayeliog) [inline packages/cardano/src/ui/util/pool-metrics.ts:27]: no need for result variable

suggested:
```
for (const [level, [min, max]] of Object.entries(saturationLevelsRangeMap) as Entries<
    typeof saturationLevelsRangeMap
  >) {
    if (inRange(saturation, min, max)) {
      return level;
    }
  }
  return SaturationLevels.Medium;
```

**Review by tommayeliog — CHANGES_REQUESTED** [review]

File: packages/cardano/src/ui/util/pool-metrics.ts (hunk 1 above)
Comment (vetalcore) [reply packages/cardano/src/ui/util/pool-metrics.ts]: done

**Review by tommayeliog — APPROVED** [review]

File: packages/staking/.storybook/__mocks__/cardano.ts [hunk 2]
```diff
@@ -1,6 +1,8 @@
 import { Cardano, util } from '@cardano-sdk/core';
+import { poolMetricsUtils } from '@lace/cardano';
```
Comment (vetalcore) [inline packages/staking/.storybook/__mocks__/cardano.ts:2]: @przemyslaw-wlodek any suggestions on how to make this one work? currently vite is aliasing  this import as well

File: packages/staking/src/features/overview/StakePoolSearch/StakePoolSearch.tsx [hunk 3]
```diff
@@ -93,29 +92,27 @@ export interface StakePoolSearchProps {
...
-  const iconColor = gettingSaturated ? '#FDC300' : saturated ? '#FF8E3C' : '#FF5470';
-  const saturationTooltip = gettingSaturated
-    ? tooltipContent.gettingSaturated
-    : saturated
-    ? tooltipContent.saturated
-    : tooltipContent.overSaturation;
-  // eslint-disable-next-line unicorn/no-null
-  const saturationIcon = notSaturated ? null : (
+  const iconColor = isOversaturated(formattedSaturation) ? '#FF5470' : saturationColoursMap[saturationLevel];
+  const saturationTooltip =
```
Comment (mchappell) [inline packages/staking/src/features/overview/StakePoolSearch/StakePoolSearch.tsx:109]: can we convert this to a switch statement func for readability?

Comment (vetalcore) [reply packages/staking/src/features/overview/StakePoolSearch/StakePoolSearch.tsx]: done

**Review by mchappell — APPROVED** [review]

**Pull request merged**